lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+
+if (!inspector || !inspector.asyncTaskScheduled) {
+  exports.setup = function() {};
+  return;
+}
+
+const hook = createHook({
+  init(asyncId, type, triggerAsyncId, resource) {
+    // It's difficult to tell which tasks will be recurring and which won't,
+    // therefore we mark all tasks as recurring. Based on the discussion
+    // in https://github.com/nodejs/node/pull/13870#discussion_r124515293,
+    // this should be fine as long as we call asyncTaskCanceled() too.
+    const recurring = true;
+    inspector.asyncTaskScheduled(type, asyncId, recurring);
+  },
+
+  before(asyncId) {
+    inspector.asyncTaskStarted(asyncId);
+  },
+
+  after(asyncId) {
+    inspector.asyncTaskFinished(asyncId);
+  },
+
+  destroy(asyncId) {
+    inspector.asyncTaskCanceled(asyncId);
+  },
+});
+
+const enable = () => {
+  if (process.bits < 64) {
+    // V8 Inspector stores task ids as (void*) pointers.
+    // async_hooks store ids as 64bit numbers.
+    // As a result, we cannot reliably translate async_hook ids to V8 async_task
+    // ids on 32bit platforms.
+    console.error(
+      'Warning: Async stack traces in debugger are not available ' +
+      `on ${process.bits}bit platforms. The feature is disabled.`);
```

Comment:
@bajtos I think this is causing the CI failures – because: `console` is a getter on the global object, which loads `process.stderr` lazily, which uses the `internal/fs` module when the stderr fd refers to a file, which loads `fs` (and not the other way around, loading `fs` and then letting that load `internal/fs`).

It’s basically the same problem as https://github.com/nodejs/node/pull/11260 / https://github.com/nodejs/node/pull/11986 …